src/Keboola/S3Extractor/S3AsyncDownloader.php

```diff
@@ -136,20 +125,37 @@ private function makeCommandPool(): CommandPool
     }
 
     /**
-     * @param ResultInterface $result
-     * @param int $index
+     * @return \Iterator|CommandInterface[]
      */
+    private function getCommands(): \Iterator
+    {
+        return makeRewindable(function (): \Iterator {
+            foreach ($this->files as $file) {
+                // Update the state of the incremental fetching
+                if ($this->state->lastTimestamp != $file->getTimestamp()) {
+                    $this->state->filesInLastTimestamp = [];
+                }
+                $this->state->lastTimestamp = max($this->state->lastTimestamp, $file->getTimestamp());
+                $this->state->filesInLastTimestamp[] = $file->getKey();
+
+                // Create command
+                yield $this->client->getCommand('getObject', $file->getParameters($this->outputDir));
+            }
+        })();
+    }
```

Comment:
A do tohto generatora, ktory vytvara  z `File` -> `Command` som pridal aj zmenu `state`.

Dalo by sa to dat aj do `fulfilled` callbacku vyssie, ale tam nemam uz k dispozicii `File` objekt.